lib/elixir/lib/access.ex

```diff
@@ -119,14 +83,67 @@ defmodule Access do
         languages: [%{name: "ELIXIR", type: :functional},
                     %{name: "C", type: :procedural}]}
 
-  See the functions `key/1`, `key!/1`, `elem/1`, and `all/0` for some of the
-  available accessors.
+  See the functions `at/1`, `key/1`, `key!/1`, `elem/1`, and `all/0` for some of the
+  available helpers.
+
+  ## Structs and Dot Syntax
+
+  As described earlier, the bracket syntax is disabled by default for struct types.
+  However, Elixir provides a "dot syntax" which can be used to traverse structs and
+  maps (with atoms for keys). It differs from bracket syntax because _it will raise for
+  non-existing keys_. This is to emphasize its usage for data structures with a _static_
+  set of keys (like struct types) whereas bracket syntax is designed for
+  data-structures with a _dynamic_ set of keys.
+
+      user = %User{name: "John"}
+      user.name
+      #=> "John"
+      
+      user.not_a_key
+      ** (KeyError) key :not_a_key not found in: %User{name: "John"}
+
+  Dot syntax does not rely on the Access behavior and as such is restricted to structs
+  and atom keys in maps.
+
+  ## The Kernel Accessors Can Use Bracket and Dot Syntax
```

Comment:
We don't use titlecase in our headers, so it should be: "The Kernel accessors can use bracket and dot syntax".